tox.ini

```diff
@@ -291,21 +292,45 @@ commands =
     bash -c 'ulimit -n 4096 || true'
     pytest -n auto --dist loadgroup --durations=20 --disable-warnings
 
-[testenv:syft.test.scenario]
-description = Syft Scenario Tests
+[testenv:stack.test.scenario.k8s]
```

Comment:
@khoaguin a couple of things for this commands

- Let's separate L0 & L2 test runs... at-least that way whenever a PR test fails, we know what is failing. Open to what you feel about this
- I was thinking of starting to make our tox tidier by leveraging the new features. We can move cluster setup code to `commands_pre` and instead of doing a bash -c, we can actually stack all the commands in a single run i.e.
```
commands_pre = 
    just delete-all start-high deploy-high
```
- We can add the teardown in `commands_post` by simply running `just delete-all`
- Now that we've added startupProbes/livenessProbes to pods, we can actually wait for the pods by simply calling the following instead of multiple `wait_for.sh`
```
kubectl wait --for=condition=ready pod --all --timeout=300s --namespace <ns> --context <ctx>
```
Infact we can add a new just command under utils group called
```
[group(...
wait-pods kube_context namespace:
  @kubectl...
```
that runs the above command or the multiple wait_for. Then we extend it for each cluster so that we have `wait-high: (wait-pods _ctx_high _ns_default)`, `wait-low` etc

- Finally, we can teardown everything in `commands_post` by calling `just delete-all`

Reply:
Also how do we do python-server testing?

Reply:
Thanks for the feedback @yashgorana. Now we have 2 separate tox and CI tasks for L0 and L2 (k8s). All using the new just commands, with commands_pre and commands_post
I made [this ticket](https://github.com/OpenMined/Heartbeat/issues/1874) to clean the whole `tox.ini`

Reply:
Working on python-server testing.